htop.c

```diff
@@ -137,9 +139,33 @@ static CommandLineSettings parseArguments(int argc, char** argv) {
             }
 
             while(pid) {
-                unsigned int num_pid = atoi(pid);
-                Hashtable_put(flags.pidWhiteList, num_pid, (void *) 1);
-                pid = strtok_r(NULL, ",", &saveptr);
+               char *endptr;
+               unsigned long num_pid;
+
+               if (pid[0] == '-') {
+                  fprintf(stderr, "Error: PID is negative \"%s\".\n", pid);
+                  goto err_next;
+               }
+
+               errno = 0;
+               num_pid = strtoul(pid, &endptr, 10);
+               if (*endptr != '\0' || errno != 0) {
+                  fprintf(stderr, "Error: failed to parse PID \"%s\"", pid);
+                  if (errno == 0)
+                     printf(".\n");
+                  else
+                     printf(": %s.\n", strerror(errno));
+                  goto err_next;
+               }
+
+               if (num_pid > UINT_MAX) {
```

Comment:
strtoul() returns `unsigned long`. `unsigned long` could be bigger than `unsigned int`. It could be produced when a user enters a number bigger than UINT_MAX.